dvc/ignore.py

```diff
@@ -174,12 +177,26 @@ def __init__(self, fs, root_dir):
         self._ignores_trie_subrepos = Trie()
 
         key = self._get_key(root_dir)
+
+        sys_root_dvcignore_path = self.fs.path.join(
```

Comment:
Just as a small note: `git`'s behaviour for global gitignore files, is to look for a global gitignore in `$XDG_CONFIG_HOME/git/ignore`.

For dvc user-level config, we use `appdirs.user_config_dir` (`system_config_dir` for system-level):
https://github.com/iterative/dvc/blob/8e5f4a8f35a15e741031bf1c9dbe4728fa55021d/dvc/config.py#L107-L116

So it could make sense to keep the user's dvcignore in `appdirs.user_config_dir(cls.APPNAME, cls.APPAUTHOR)` (note: no leading `.`)

Reply:
@dtrifiro Thank you for your review!

Happy to make that change but IMHO `~/.dvcignore` is more intuitive to maintain than `~/Library/Application Support/dvc/dvcignore` (macOS).

I understand that the global config options get stored in `~/Library/Application Support/dvc/config` but something to keep in mind is that the documented way of setting and reading global config options is through the command `dvc config --global` rather than maintaining the config file itself.

Lmk what you think, I'm happy to make it go either way.

Reply:
IMO it would be better to handle this the way that git does - you tell git where to look for the file via the `core.excludesfile` config option.

We should be adding a similar config option in DVC, so the user can do `dvc config --global core.excludesfile $HOME/.dvcignore` if that's where they want it to go. If the user has not set the option, it should default to looking in the proper DVC user config directories per-platform (`~/Library/Application Support` on macos, XDG config dirs on linux, and `%localappdata%` on windows).

Reply:
Also note that by default we need to check the global (user) level config location first. But we should also check the system level location as well if there is no global level .dvcignore.

Reply:
@pmrowla okay cool thanks. I'll update the code accordingly!